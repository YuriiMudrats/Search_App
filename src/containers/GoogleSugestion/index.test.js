import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";

import { GoogleSuggestion } from './index.jsx';


const googleLists = [
  'world cup',
  'world map',
  'world of tanks',
  'worldtimebuddy',
  'world cup groups',
  'world of warcraft',
  'world cup standings',
  'world news',
  'world cup results',
  'world cup matches'
];

const renderComponent = (props = {}) =>
  shallow(<GoogleSuggestion googleLists={googleLists} {...props} />);

describe('<GoogleSuggestion />', () => {
  it('should render null', () => {
    const renderedComponent = shallow(<GoogleSuggestion googleLists={[]} />);;
    expect(renderedComponent.type()).toEqual(null);
  });
  it('should render an <ul> tag', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.type()).toEqual('ul');
  });
  it('<ul> should have className attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('className')).toBeDefined();
  });
  it('should have 10 of <li> tags', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.find('li').length).toEqual(10);
  });
  it('<li> should have className attribute', () => {
    const renderedComponent = renderComponent();
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });

});
