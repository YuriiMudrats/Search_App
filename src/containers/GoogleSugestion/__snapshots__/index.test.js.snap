// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleSuggestion /> <li> should have className attribute 1`] = `
<ul
  className="search__list"
>
  <li
    className="search__list__item"
    key="world cup0"
    onClick={[Function]}
  >
    world cup
  </li>
  <li
    className="search__list__item"
    key="world map1"
    onClick={[Function]}
  >
    world map
  </li>
  <li
    className="search__list__item"
    key="world of tanks2"
    onClick={[Function]}
  >
    world of tanks
  </li>
  <li
    className="search__list__item"
    key="worldtimebuddy3"
    onClick={[Function]}
  >
    worldtimebuddy
  </li>
  <li
    className="search__list__item"
    key="world cup groups4"
    onClick={[Function]}
  >
    world cup groups
  </li>
  <li
    className="search__list__item"
    key="world of warcraft5"
    onClick={[Function]}
  >
    world of warcraft
  </li>
  <li
    className="search__list__item"
    key="world cup standings6"
    onClick={[Function]}
  >
    world cup standings
  </li>
  <li
    className="search__list__item"
    key="world news7"
    onClick={[Function]}
  >
    world news
  </li>
  <li
    className="search__list__item"
    key="world cup results8"
    onClick={[Function]}
  >
    world cup results
  </li>
  <li
    className="search__list__item"
    key="world cup matches9"
    onClick={[Function]}
  >
    world cup matches
  </li>
</ul>
`;

exports[`<GoogleSuggestion /> should be equal to snapshot 1`] = `
<ul
  className="search__list"
>
  <li
    className="search__list__item"
    key="world cup0"
    onClick={[Function]}
  >
    world cup
  </li>
  <li
    className="search__list__item"
    key="world map1"
    onClick={[Function]}
  >
    world map
  </li>
  <li
    className="search__list__item"
    key="world of tanks2"
    onClick={[Function]}
  >
    world of tanks
  </li>
  <li
    className="search__list__item"
    key="worldtimebuddy3"
    onClick={[Function]}
  >
    worldtimebuddy
  </li>
  <li
    className="search__list__item"
    key="world cup groups4"
    onClick={[Function]}
  >
    world cup groups
  </li>
  <li
    className="search__list__item"
    key="world of warcraft5"
    onClick={[Function]}
  >
    world of warcraft
  </li>
  <li
    className="search__list__item"
    key="world cup standings6"
    onClick={[Function]}
  >
    world cup standings
  </li>
  <li
    className="search__list__item"
    key="world news7"
    onClick={[Function]}
  >
    world news
  </li>
  <li
    className="search__list__item"
    key="world cup results8"
    onClick={[Function]}
  >
    world cup results
  </li>
  <li
    className="search__list__item"
    key="world cup matches9"
    onClick={[Function]}
  >
    world cup matches
  </li>
</ul>
`;
